src/Frontend/IAST/Abs.hs

```diff
@@ -0,0 +1,33 @@
+module Frontend.IAST.Abs where
+
+import Data.Complex (Complex)
+
+data Id = Id
+    { pos :: (Int, Int)
+    , name :: String 
+    }
+  deriving (Eq, Show, Read)
+  
+newtype Program = Progr [Toplevel]
+  deriving (Eq, Show, Read) 
+
+data Basis
+    = Z
+    | X
+    | None
+  deriving (Eq, Show, Read) 
+data Toplevel
+    = ToplF Id [Id] Expr
+    | Topl Basis (Complex Double) [Id] Expr
+  deriving (Eq, Show, Read)
+  
+data Expr
+    = Var Id
+    | Tup Expr [Expr]
```

Comment:
Convert this to a single expr-list. The reason for it being two expressions in the parser is the requirement for tuples to include at least 2 elements.
```suggestion
    | Tup [Expr]
```

Reply:
> Or even use a non-empty list here, or a sized list.